app/src/main/java/com/jwisozk/flightreviews/ReviewAdapter.kt

```diff
@@ -0,0 +1,131 @@
+        penultimateIndex = labelArray.lastIndex - 1
+        lastIndex = labelArray.lastIndex
+    }
+
+    inner class ButtonHolder(@NonNull itemView: View) : RecyclerView.ViewHolder(itemView) {
+        fun bind(@NonNull itemView: View) {
+            val button = itemView.findViewById<Button>(R.id.submitButton)
+            button.setOnClickListener {
+                if (editText.text.isNotEmpty())
+                    viewModel.setParameters(textParam = editText.text.toString())
+                if (param != null) {
+                    viewModel.refreshData()
+                }
+            }
+        }
+    }
+
+    inner class RatingBarHolder(@NonNull itemView: ViewGroup) : RecyclerView.ViewHolder(itemView) {
+        fun bind(@NonNull itemView: View, position: Int) {
+            if (position == penultimateIndex) {
+                val checkBox = itemView.findViewById<CheckBox>(R.id.foodCheckBox)
+                checkBox.visibility = View.VISIBLE
+                checkBox.setOnCheckedChangeListener { _, isChecked ->
+                    if (isChecked) viewModel.setParameters(foodParam = null)
+                }
+            }
+            val ratingBar = itemView.findViewById<RatingBar>(R.id.ratingBar)
+            ratingBar.onRatingBarChangeListener =
+                RatingBar.OnRatingBarChangeListener { _, rating, _ ->
+                    run {
+                        when (position) {
+                            0 -> viewModel.onOverallRatingChanged(rating, ParamEnum.PEOPLE)
+                            1 -> viewModel.onOverallRatingChanged(rating, ParamEnum.AIRCRAFT)
+                            2 -> viewModel.onOverallRatingChanged(rating, ParamEnum.SEAT)
+                            3 -> viewModel.onOverallRatingChanged(rating, ParamEnum.CREW)
+                            4 -> {
+                                param?.food?.let {
+                                    viewModel.onOverallRatingChanged(rating, ParamEnum.FOOD)
+                                }
+
+                            }
+                        }
+                    }
+                }
+            val textView = itemView.findViewById<TextView>(R.id.labelRatingBar)
+            textView.text = labelArray[position]
+        }
+    }
+
+    inner class EditTextHolder(@NonNull itemView: ViewGroup) : RecyclerView.ViewHolder(itemView) {
+        fun bind(@NonNull itemView: View, position: Int) {
+            editText = itemView.findViewById(R.id.feedbackEditText)
+            val textView = itemView.findViewById<TextView>(R.id.labelEditText)
+            textView.text = labelArray[position]
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return when (position) {
+            labelArray.size -> R.layout.button
+            lastIndex -> R.layout.edit_text
+            else -> R.layout.rating_bar
+        }
+    }
+
+    @NonNull
+    override fun onCreateViewHolder(
+        @NonNull parent: ViewGroup,
```

Comment:
let's pass `LayoutInflater` here as well, so we don't invoke context inside `recycler`